account-integrations/safe/src/SafeWebAuthnPlugin.sol

```diff
@@ -67,35 +67,92 @@ contract SafeWebAuthnPlugin is BaseAccount {
         return _publicKey;
     }
 
+    struct LocalVarStruct {
+        bytes1 authenticatorDataFlagMask;
+        uint256 clientChallengeDataOffset;
+    }
+
     function _validateSignature(
         UserOperation calldata userOp,
         bytes32 userOpHash
     ) internal override returns (uint256 validationData) {
-        {
+        bytes calldata authenticatorData;
+        bytes calldata clientData;
+        uint256[2] calldata signature;
+        uint256[2] calldata pubKey;
+        LocalVarStruct memory s;
+
+        {            
+            // parse length of all fixed-length params (including length)
+            uint i = 0;
+            uint dataLen = 32;
+            uint256 paramLen = abi.decode(userOp.signature[i:i+dataLen], (uint256));
```

Comment:
Out of interest, why do we need to set param length here? I can't see it used again until it is set to get the authenticator data: `paramLen = abi.decode(userOp.signature[i:i + dataLen], (uint256));`

Reply:
Updated to use the read paramLen (with comments).